lib/vfscore/syscalls.c

```diff
@@ -839,20 +839,23 @@ sys_rename(char *src, char *dest)
 	vn_lock(dvp1);
 
 	dvp2 = ddp2->d_vnode;
-	vn_lock(dvp2);
+	if (dvp1 != dvp2) {
+		vn_lock(dvp2);
 
-	/* Source and destination directions should be writable) */
-	if ((error = vn_access(dvp1, VWRITE)) != 0)
-	    goto err3;
-	if ((error = vn_access(dvp2, VWRITE)) != 0)
-	    goto err3;
+		/* The source and dest must be same file system */
```

Comment:
```suggestion
		/* The source and destination must be in the same file system */
```

Not sure about the meaning of the previous comment, so I amended it a bit.